### ISSUE TITLE
Set-PSReadlineOption -HistorySaveStyle SaveAtExit not working on Windows 10 build 10162

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/PSReadLine#262]
This option hasn't been working in the last couple of Windows 10 builds dating back to build 10130 I think.  I can workaround this for now by not using this option and letting it save incrementally.

**Comment by gwojan** [comment]
I don't think it works correctly on Windows 7 either. I recently started using -SaveAtExit with the latest builds and my history wasn't saved at all.

**Comment by rkeithhill** [comment]
@gwojan  Thanks for confirming the issue.

**Comment by daxian-dbw** [comment]
@rkeithhill Can you please confirm if this is still an issue with the latest PSReadLine module?

**Comment by ghost** [comment]
This issue is closed because it has been marked as requiring author feedback but has not had any activity for **7 days**. If you think the issue is still relevant, please reopen and provide your feedback.

**Comment by rkeithhill** [comment]
@daxian-dbw AFAICT nothing is saved in the history file if you use `SaveAtExit`. 🤷‍♂️

**Comment by daxian-dbw** [comment]
@rkeithhill Thanks for checking and confirming.

**Comment by puxu-msft** [comment]
same issue, 2.2.5 and win11.

**Comment by Shuangcheng-Ni** [comment]
I've tested this command with PowerShell 5.1 and 7.3 on Windows 11. Is this the expected behavior?
| PowerShell Version \ exit by | `exit` command | close console window |
|:---------------------:|:-----------------:|:-------------------------:|
| 5.1 | not save | not save |
| 7.3 | not save | save |

**Comment by StevenBucher98** [comment]
I tested this and have had the following behaviors with PSReadline 2.3.3 and Windows 11:

PowerShell Version \ exit by | exit command | close console window | close console window Terminal
-- | -- | -- | --
5.1 | not save | save | save 
7.3 | not save | save | save

Personally I would expect history to be saved when `exit` is typed, unsure if this is a technical limitation @daxian-dbw?

Also I get no history saved when I do `SaveAtExit` on Mac.
